src/lib/entity-state.ts

```diff
@@ -491,3 +532,33 @@ function elvis(object: any, path: string): any | undefined {
 function asArray<T>(input: T | T[]): T[] {
   return Array.isArray(input) ? input : [input];
 }
+
+/**
+ * Limits a number to the given boundaries
+ * @param value The input value
+ * @param min The minimum value
+ * @param max The maximum value
+ */
+function clamp(value: number, min: number, max: number): number {
+  return Math.min(max, Math.max(min, value));
+}
+
+/**
+ * Uses the clamp function is wrap is false.
+ * Else it wrap to the max or min value respectively.
+ * @param wrap Flag to indicate if value should be wrapped
+ * @param value The input value
+ * @param min The minimum value
+ * @param max The maximum value
+ */
+function wrapOrClamp(wrap: boolean, value: number, min: number, max: number): number {
+  if (!wrap) {
```

Comment:
I don't mind if you don't want to, but I prefer writing it like that:
```typescript
let newValue = value;

if (!wrap) {
  newValue = ...;
} else if (...) {
  ...
} else if (...) {
  ...
}

return newValue;
```

Reply:
I would like to keep mine.